### PR TITLE
sayonara:  init at 1.0.0-git5-20180115

### LINKED ISSUE #34024: Packaging request: Sayonara music player
## Issue description

Package http://sayonara-player.com/index.php

Why:
- it has an interesting library management UI
- has useful filesystem-based library view
- generally nice-looking

Dependencies:
- quoting http://sayonara-player.com/howto-build.php
```
sudo apt-get install git cmake libgstreamer-plugins-bad-1.0-dev libgstreamer-plugins-base1.0-dev libgstreamer1.0-dev libtag1-dev pkg-config qtbase5-dev qttools5-dev qttools5-dev-tools zlib1g-dev 
```
- summary:  gstreamer, libtag, QT 5, zlib

### PR DESCRIPTION
**Pull request by deepfire**
###### Motivation for this change
Fix #34024 

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by joachifm** [comment]
@GrahamcOfBorg build sayonara

File: pkgs/applications/audio/sayonara/default.nix [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+in
+stdenv.mkDerivation {
+  name = "sayonara-player-${version}";
+
+  src = fetchurl {
+    url = "https://sayonara-player.com/sw/sayonara-player-${version}.tar.gz";
+    sha256 = "1fl7zplnrrvbv1xm4g348bpd46jj39jvbm808hyjjq92i64wqg37";
+  };
+
+  buildInputs = [ cmake qt5.qtbase qt5.qttools zlib taglib pkgconfig pcre gst_all_1.gstreamer gst_all_1.gst-plugins-base gst_all_1.gst-plugins-good gst_all_1.gst-plugins-ugly ];
```
Comment (joachifm) [inline pkgs/applications/audio/sayonara/default.nix]: At least `cmake` and `pkgconfig` can be made into `nativeBuildInputs`

File: pkgs/applications/audio/sayonara/default.nix [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  # [ 65%] Building CXX object src/Components/Engine/CMakeFiles/say_comp_engine.dir/AbstractPipeline.cpp.o
+  # /tmp/nix-build-sayonara-player-1.0.0-git5-20180115.drv-0/sayonara-player/src/Components/Engine/AbstractPipeline.cpp:28:32: fatal error: gst/app/gstappsink.h: No such file or directory
+  #  #include <gst/app/gstappsink.h>
+  NIX_CFLAGS_COMPILE = "-I${gst_all_1.gst-plugins-base.dev}/include/gstreamer-1.0";
+
+  configurePhase = ''
+    mkdir build
+    cd build
+    runHook preConfigure
+    cmake .. \
```
Comment (joachifm) [inline pkgs/applications/audio/sayonara/default.nix]: What about setting `cmakeFlags` ?

Comment (deepfire) [reply pkgs/applications/audio/sayonara/default.nix]: @joachifm, do you have any particular suggestions?

Comment (joachifm) [reply pkgs/applications/audio/sayonara/default.nix]: as in `cmakeFlags = [ ... ];` & relying on the standard configure phase definition (potentially using `preConfigure` to create & enter `build`.

File: pkgs/applications/audio/sayonara/default.nix (hunk 1 above)
Comment (deepfire) [reply pkgs/applications/audio/sayonara/default.nix]: Done..

**Comment by deepfire** [comment]
@joachifm, all fixed now, I believe.

File: pkgs/applications/audio/sayonara/default.nix [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+      --replace "/usr/share" "$out/share"
+  '';
+
+  # [ 65%] Building CXX object src/Components/Engine/CMakeFiles/say_comp_engine.dir/AbstractPipeline.cpp.o
+  # /tmp/nix-build-sayonara-player-1.0.0-git5-20180115.drv-0/sayonara-player/src/Components/Engine/AbstractPipeline.cpp:28:32: fatal error: gst/app/gstappsink.h: No such file or directory
+  #  #include <gst/app/gstappsink.h>
+  NIX_CFLAGS_COMPILE = "-I${gst_all_1.gst-plugins-base.dev}/include/gstreamer-1.0";
+
+  cmakeFlags = [
+    "-DCMAKE_BUILD_TYPE=Release"
```
Comment (joachifm) [inline pkgs/applications/audio/sayonara/default.nix]: This is set by default in the cmake setup hook, is it really necessary?

Comment (deepfire) [reply pkgs/applications/audio/sayonara/default.nix]: Fixed..

**Comment by joachifm** [comment]
@GrahamcOfBorg build sayonara

**Comment by deepfire** [comment]
@joachifm, do you feel there are any other blockers remaining?

**Pull request merged**